client/state/connectionMiddleware/middleware.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * External dependencies
+ */
+import uniqueId from 'lodash/utility/uniqueId';
+
+/**
+ * Internal dependencies
+ */
+import {
+	isOffline,
+	isOnline
+} from 'state/application/selectors';
+import {
+	OFFLINE_QUEUE_ACTION,
+	CONNECTION_RESTORED,
+	OFFLINE_QUEUE_REMOVE,
+} from 'state/action-types';
+const ACTION_TIMEOUT = 10000;
+
+const connections = {};
+
+let actionTimeout = null;
+
+//Prob should be in actions
+function queueAction( action ) {
+	return {
```

Comment:
Proposed improvement:
If `registerConnection` was set with certain option (likesquash:true )` then only last call is fired after reconnecting.
That way, we dont do 120 fetches at once.
This would only make sense for GET like requests.